### PR TITLE
convert the csv datatable to a shapefile

### PR DESCRIPTION
**Pull request by Christian-Mielke**
There have been minor problems and issues with the utf8 encoding and jupyter when running the convert function, which I fixed so hopefully it works on your machines. Otherwise I'd highly suggest to drop Jupyter notebooks in favour of pure python code/ pyfiles.

### CODE REVIEW

File: Convert ePRTR data to CLRTAP LPS.ipynb [hunk 1]
```diff
@@ -111,7 +111,10 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "data.to_csv('output/Convert ePRTR data to CLRTAP LPS.csv', sep='|', index=False, columns=['LPS', 'GNFR', 'E-PRTR/PRTR Facility ID', 'Height class', 'Longitude (deg)', 'Latitude (deg)', 'NOx (as NO2) (kt)', 'NMVOC (kt)', 'SOx (as SO2) (kt)', 'NH3 (kt)', 'PM2.5 (kt)', 'PM10 (kt)', 'CO (kt)', 'Pb (t)', 'Cd (t)', 'Hg (t)', 'PCDD/ PCDF (dioxins/ furans)(g I-Teq)', 'PAHs (t)', 'HCB (kg)', 'PCBs (kg)'])"
+    "data.to_csv('output/Convert ePRTR data to CLRTAP LPS.csv', sep='|', index=False, columns=['LPS', 'GNFR', 'E-PRTR/PRTR Facility ID', 'Height class', 'Longitude (deg)', 'Latitude (deg)', 'NOx (as NO2) (kt)', 'NMVOC (kt)', 'SOx (as SO2) (kt)', 'NH3 (kt)', 'PM2.5 (kt)', 'PM10 (kt)', 'CO (kt)', 'Pb (t)', 'Cd (t)', 'Hg (t)', 'PCDD/ PCDF (dioxins/ furans)(g I-Teq)', 'PAHs (t)', 'HCB (kg)', 'PCBs (kg)'])\n",
+    "#Convert the csv file to a shapefile using the prtrgpd function\n",
```
Comment (kevin-hausmann) [inline Convert ePRTR data to CLRTAP LPS.ipynb:115]: Please put this into a new cell to make it optional

File: Convert ePRTR data to CLRTAP LPS.ipynb [hunk 2]
```diff
@@ -131,7 +134,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.6"
+   "version": "3.8.5"
```
Comment (kevin-hausmann) [inline Convert ePRTR data to CLRTAP LPS.ipynb:137]: Again, please do not commit changes like this. These are envoironment dependent changes that should not be part of the commit. You might need to use a texteditor to revert the changes done by Jupyter notebooks.

File: prtrgpd.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#small function to convert the lps csv output to a shapefile for GIS Display and Work
+import geopandas as gpd
+import pandas as pd
+
+def convert_output_to_shp(input,delimit='|',header=0):
+    data=pd.read_csv(input,delimiter=delimit,header=0,encoding='utf8')
+    bname=input.split('.')[0]
```
Comment (kevin-hausmann) [inline prtrgpd.py:7]: Why is this call `bname`, shouldn't it be `filename`?

File: prtrgpd.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    locations=gpd.GeoDataFrame(data,geometry=gpd.points_from_xy(data['Longitude (deg)'],data['Latitude (deg)']),crs='epsg:4326')#watch out for the Lat Long kwds
```
Comment (kevin-hausmann) [inline prtrgpd.py:8]: I do not understand this comment, what does kwds mean?

File: prtrgpd.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#small function to convert the lps csv output to a shapefile for GIS Display and Work
+import geopandas as gpd
+import pandas as pd
+
+def convert_output_to_shp(input,delimit='|',header=0):
```
Comment (kevin-hausmann) [inline prtrgpd.py:5]: Please activate Python linting and format checking in your editor. Spyder also supports this and it helps to make the Python code more readable. See https://docs.spyder-ide.org/current/pylint.html

**Review by kevin-hausmann — DISMISSED** [review]
Looks good, this is really useful! I opened the resulting files and it work well. However, please adjust a new thing as requested below - thanks!

**Comment by Christian-Mielke** [comment]
Ok I have new ideas for features to put in which are e.g.:
1. Also supply shapefiles as cutline for each Fed. State to be used for Clipping LPS at federal State Level.
2. Spatial Interpolation to a continous raster grid e.g. 5x5km for display

**Comment by kevin-hausmann** [comment]
> Ok I have new ideas for features to put in which are e.g.:
>     1. Also supply shapefiles as cutline for each Fed. State to be used for Clipping LPS at federal State Level.

Sounds good.
 
>     2. Spatial Interpolation to a continous raster grid e.g. 5x5km for display

Not sure I understand this one. What is the reasoning behind this, i.e. what is the raster used for? Thanks!

**Comment by AnnaZagorski** [comment]
I also think that that an Auto-Export-Function for LPS Data is very useful. Then we wait for the revised version in a new pull-request and with that I close this pull-request.

**Comment by kevin-hausmann** [comment]
Hi @AnnaZagorski - pull requests do not need to be closed to be changed. @Christian-Mielke can simple push an additional commit to address the requested changes to the underlying branch and then re-request our review.

**Comment by AnnaZagorski** [comment]
ok, then we will wait for the changes.

**Comment by Christian-Mielke** [comment]
Ok I have redone everything according to the review and re-requested the review. Hopefully this is the right way to do that

**Comment by Christian-Mielke** [comment]
Do _Not_ merge this commit rather only merge chimney as it already includes shapefile branch

**Review by kevin-hausmann — APPROVED** [review]

**Review by Christian-Mielke — COMMENTED** [review]
Alles okay, werde dast chimney genau so abwandeln.

**Comment by kevin-hausmann** [comment]
Nur so für die Zukunft:
@AnnaZagorski  - warum machst du PRs zu, wenn sie noch nicht gemergt oder fertig sind?
@Christian-Mielke  - besser wäre folgendes Vorgehen: 1. Diesen PR mergen, 2. Stack height PR rebasen (falls es überhaupt einen Konflikt gibt)

**Comment by Christian-Mielke** [comment]
Okay then we are doing this incrementally via rebase.

**Review by AnnaZagorski — COMMENTED** [review]
this is ok!

**Review by AnnaZagorski — APPROVED** [review]
